app/src/main/java/pl/bk20/forest/core/data/source/DayDao.kt

```diff
@@ -9,7 +9,7 @@ import java.time.LocalDate
 @Dao
 interface DayDao {
 
-    @Query("SELECT steps / goal as treeAmount FROM day WHERE steps >= goal")
+    @Query("SELECT SUM(steps/goal) FROM day WHERE steps >= goal")
```

Comment:
You can drop the `WHERE steps >= goal`. The `SUM` will only count collected trees. This is because we're dividing integers here.

Reply:
At least I think so. Please confirm it still works as expected.